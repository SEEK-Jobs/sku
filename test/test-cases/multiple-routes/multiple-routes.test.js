const path = require('path');
const dirContentsToObject = require('../../utils/dirContentsToObject');
const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const waitForUrls = require('../../utils/waitForUrls');
const { getAppSnapshot } = require('../../utils/appSnapshot');
const startAssetServer = require('../../utils/assetServer');
const appDir = path.resolve(__dirname, 'app');

const targetDirectory = `${appDir}/dist`;

describe('multiple-routes', () => {
  describe('start', () => {
    const devServerUrl = `http://localhost:8202`;
    let server;

    beforeAll(async () => {
      server = await runSkuScriptInDir('start', appDir);
      await waitForUrls(devServerUrl);
    });

    afterAll(() => {
      server.kill();
    });

    it('should render home page correctly', async () => {
      const snapshot = await getAppSnapshot(devServerUrl);
      expect(snapshot).toMatchSnapshot();
    });

    it('should render details page correctly', async () => {
      const snapshot = await getAppSnapshot(`${devServerUrl}/details/123`);
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('test', () => {
    it('should handle dynamic imports in tests', async () => {
      const { childProcess } = await runSkuScriptInDir('test', appDir);
      expect(childProcess.exitCode).toEqual(0);
    });
  });

  describe('build', () => {
    let closeAssetServer;

    beforeAll(async () => {
      await runSkuScriptInDir('build', appDir);
      closeAssetServer = await startAssetServer(4004, targetDirectory, [
        { source: '/', destination: '/production/au/index.html' },
      ]);
    });

    afterAll(() => {
      closeAssetServer();
    });

    it('should create valid app', async () => {
      const app = await getAppSnapshot('http://localhost:4004');
      expect(app).toMatchSnapshot();
    });

    it('should generate the expected files', async () => {
      const files = await dirContentsToObject(targetDirectory);
      expect(files).toMatchSnapshot();
    });
  });
});
