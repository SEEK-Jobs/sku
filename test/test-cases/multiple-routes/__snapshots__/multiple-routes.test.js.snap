// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiple-routes build should create valid app 1`] = `
SCRIPTS: Array [
  "http://localhost:4004/runtime-1b8326c78bf3df922acc.js",
  "http://localhost:4004/vendors~main-f9d0092723c71e65e66e.js",
  "http://localhost:4004/main-648b7c92a3febada80f7.js",
  "http://localhost:4004/handlers-Home-65fb4288f1865855b7dd.js",
]
CSS: Array [
  "http://localhost:4004/handlers-Home-62bf94bf0fd2ba3c2aa2.css",
]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="handlers-Home"
          rel="stylesheet"
          href="styles[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
          crossorigin="anonymous"
    >
    <link data-chunk="handlers-Home"
          rel="preload"
          as="script"
          href="scripts[3]"
          crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <h1 class="_1QX0Hyt">
        Welcome to the Home page -
        au
      </h1>
    </div>
    <script>
      window.__SKU_CLIENT_CONTEXT__ = {"site":"au"};
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
            crossorigin="anonymous"
    >
      [2]
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="handlers-Home"
            src="scripts[3]"
            crossorigin="anonymous"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`multiple-routes build should generate the expected files 1`] = `
Object {
  "AsyncComponent-c44d4a8a700c39761853.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "handlers-Details-0e82719ea3ab71aafb06.css": ._91MnZhe {
  color: #00f;
}
,
  "handlers-Details-63ddae3291ad6b8dde4d.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "handlers-Home-62bf94bf0fd2ba3c2aa2.css": ._1QX0Hyt {
  color: red;
}
,
  "handlers-Home-65fb4288f1865855b7dd.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "main-648b7c92a3febada80f7.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "production/au/details/:id/index.html": SCRIPTS: Array [
    "http://localhost:4004/runtime-1b8326c78bf3df922acc.js",
    "http://localhost:4004/vendors~main-f9d0092723c71e65e66e.js",
    "http://localhost:4004/main-648b7c92a3febada80f7.js",
    "http://localhost:4004/handlers-Details-63ddae3291ad6b8dde4d.js",
    "http://localhost:4004/AsyncComponent-c44d4a8a700c39761853.js",
  ]
CSS: Array [
    "http://localhost:4004/handlers-Details-0e82719ea3ab71aafb06.css",
  ]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="handlers-Details"
          rel="stylesheet"
          href="styles[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
          crossorigin="anonymous"
    >
    <link data-chunk="handlers-Details"
          rel="preload"
          as="script"
          href="scripts[3]"
          crossorigin="anonymous"
    >
    <link data-chunk="AsyncComponent"
          rel="preload"
          as="script"
          href="scripts[4]"
          crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <h1 class="_91MnZhe">
        Welcome to the Details page -
        au
        <span>
          Some special async content
        </span>
      </h1>
    </div>
    <script>
      window.__SKU_CLIENT_CONTEXT__ = {"site":"au"};
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
            crossorigin="anonymous"
    >
      [1,0]
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="handlers-Details"
            src="scripts[3]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="AsyncComponent"
            src="scripts[4]"
            crossorigin="anonymous"
    >
    </script>
  </body>
</html>,
  "production/au/index.html": SCRIPTS: Array [
    "http://localhost:4004/runtime-1b8326c78bf3df922acc.js",
    "http://localhost:4004/vendors~main-f9d0092723c71e65e66e.js",
    "http://localhost:4004/main-648b7c92a3febada80f7.js",
    "http://localhost:4004/handlers-Home-65fb4288f1865855b7dd.js",
  ]
CSS: Array [
    "http://localhost:4004/handlers-Home-62bf94bf0fd2ba3c2aa2.css",
  ]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="handlers-Home"
          rel="stylesheet"
          href="styles[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
          crossorigin="anonymous"
    >
    <link data-chunk="handlers-Home"
          rel="preload"
          as="script"
          href="scripts[3]"
          crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <h1 class="_1QX0Hyt">
        Welcome to the Home page -
        au
      </h1>
    </div>
    <script>
      window.__SKU_CLIENT_CONTEXT__ = {"site":"au"};
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
            crossorigin="anonymous"
    >
      [2]
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="handlers-Home"
            src="scripts[3]"
            crossorigin="anonymous"
    >
    </script>
  </body>
</html>,
  "production/nz/details/:id/index.html": SCRIPTS: Array [
    "http://localhost:4004/runtime-1b8326c78bf3df922acc.js",
    "http://localhost:4004/vendors~main-f9d0092723c71e65e66e.js",
    "http://localhost:4004/main-648b7c92a3febada80f7.js",
    "http://localhost:4004/handlers-Details-63ddae3291ad6b8dde4d.js",
    "http://localhost:4004/AsyncComponent-c44d4a8a700c39761853.js",
  ]
CSS: Array [
    "http://localhost:4004/handlers-Details-0e82719ea3ab71aafb06.css",
  ]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="handlers-Details"
          rel="stylesheet"
          href="styles[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
          crossorigin="anonymous"
    >
    <link data-chunk="handlers-Details"
          rel="preload"
          as="script"
          href="scripts[3]"
          crossorigin="anonymous"
    >
    <link data-chunk="AsyncComponent"
          rel="preload"
          as="script"
          href="scripts[4]"
          crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <h1 class="_91MnZhe">
        Welcome to the Details page -
        nz
        <span>
          Some special async content
        </span>
      </h1>
    </div>
    <script>
      window.__SKU_CLIENT_CONTEXT__ = {"site":"nz"};
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
            crossorigin="anonymous"
    >
      [1,0]
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="handlers-Details"
            src="scripts[3]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="AsyncComponent"
            src="scripts[4]"
            crossorigin="anonymous"
    >
    </script>
  </body>
</html>,
  "production/nz/index.html": SCRIPTS: Array [
    "http://localhost:4004/runtime-1b8326c78bf3df922acc.js",
    "http://localhost:4004/vendors~main-f9d0092723c71e65e66e.js",
    "http://localhost:4004/main-648b7c92a3febada80f7.js",
    "http://localhost:4004/handlers-Home-65fb4288f1865855b7dd.js",
  ]
CSS: Array [
    "http://localhost:4004/handlers-Home-62bf94bf0fd2ba3c2aa2.css",
  ]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="handlers-Home"
          rel="stylesheet"
          href="styles[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
          crossorigin="anonymous"
    >
    <link data-chunk="handlers-Home"
          rel="preload"
          as="script"
          href="scripts[3]"
          crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <h1 class="_1QX0Hyt">
        Welcome to the Home page -
        nz
      </h1>
    </div>
    <script>
      window.__SKU_CLIENT_CONTEXT__ = {"site":"nz"};
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
            crossorigin="anonymous"
    >
      [2]
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="handlers-Home"
            src="scripts[3]"
            crossorigin="anonymous"
    >
    </script>
  </body>
</html>,
  "runtime-1b8326c78bf3df922acc.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "vendors~main-f9d0092723c71e65e66e.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;

exports[`multiple-routes start should render details page correctly 1`] = `
SCRIPTS: Array [
  "/runtime.js",
  "/vendors~devServerOnly~main.js",
  "/vendors~main.js",
  "/main.js",
  "/handlers-Details.js",
  "/AsyncComponent.js",
]
CSS: Array [
  "/handlers-Details.css",
]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="handlers-Details"
          rel="stylesheet"
          href="styles[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[3]"
    >
    <link data-chunk="handlers-Details"
          rel="preload"
          as="script"
          href="scripts[4]"
    >
    <link data-chunk="AsyncComponent"
          rel="preload"
          as="script"
          href="scripts[5]"
    >
  </head>
  <body>
    <div id="app">
      <h1 class="Details__root___91MnZhe">
        Welcome to the Details page -
        au
        <span>
          Some special async content
        </span>
      </h1>
    </div>
    <script>
      window.__SKU_CLIENT_CONTEXT__ = {"site":"au"};
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
    >
      ["handlers-Details","AsyncComponent"]
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[3]"
    >
    </script>
    <script async
            data-chunk="handlers-Details"
            src="scripts[4]"
    >
    </script>
    <script async
            data-chunk="AsyncComponent"
            src="scripts[5]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: 
     <div id="app">
       <h1 class="Details__root___91MnZhe">
         Welcome to the Details page -
-        au
+        auID: 123
         <span>
           Some special async content
         </span>

`;

exports[`multiple-routes start should render home page correctly 1`] = `
SCRIPTS: Array [
  "/runtime.js",
  "/vendors~devServerOnly~main.js",
  "/vendors~main.js",
  "/main.js",
  "/handlers-Home.js",
]
CSS: Array [
  "/handlers-Home.css",
]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="handlers-Home"
          rel="stylesheet"
          href="styles[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[3]"
    >
    <link data-chunk="handlers-Home"
          rel="preload"
          as="script"
          href="scripts[4]"
    >
  </head>
  <body>
    <div id="app">
      <h1 class="Home__root___1QX0Hyt">
        Welcome to the Home page -
        au
      </h1>
    </div>
    <script>
      window.__SKU_CLIENT_CONTEXT__ = {"site":"au"};
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
    >
      ["handlers-Home"]
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[3]"
    >
    </script>
    <script async
            data-chunk="handlers-Home"
            src="scripts[4]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;
